diff --git a/ldm/models/diffusion/ddpm.py b/ldm/models/diffusion/ddpm.py
index 3350c03..9ba2919 100644
--- a/ldm/models/diffusion/ddpm.py
+++ b/ldm/models/diffusion/ddpm.py
@@ -17,7 +17,7 @@ from functools import partial
 import itertools
 from tqdm import tqdm
 from torchvision.utils import make_grid
-from pytorch_lightning.utilities.distributed import rank_zero_only
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
 from omegaconf import ListConfig
 
 from ldm.util import log_txt_as_img, exists, default, ismap, isimage, mean_flat, count_params, instantiate_from_config
diff --git a/ldm/modules/diffusionmodules/util.py b/ldm/modules/diffusionmodules/util.py
index daf35da..0238e6e 100644
--- a/ldm/modules/diffusionmodules/util.py
+++ b/ldm/modules/diffusionmodules/util.py
@@ -140,7 +140,7 @@ class CheckpointFunction(torch.autograd.Function):
     def backward(ctx, *output_grads):
         ctx.input_tensors = [x.detach().requires_grad_(True) for x in ctx.input_tensors]
         with torch.enable_grad(), \
-                torch.cuda.amp.autocast(**ctx.gpu_autocast_kwargs):
+                torch.amp.autocast("cuda", **ctx.gpu_autocast_kwargs):
             # Fixes a bug where the first op in run_function modifies the
             # Tensor storage in place, which is not allowed for detach()'d
             # Tensors.
diff --git a/ldm/util.py b/ldm/util.py
index 9ede259..b8fd99f 100644
--- a/ldm/util.py
+++ b/ldm/util.py
@@ -10,7 +10,7 @@ from PIL import Image, ImageDraw, ImageFont
 
 def autocast(f):
     def do_autocast(*args, **kwargs):
-        with torch.cuda.amp.autocast(enabled=True,
+        with torch.amp.autocast("cuda", enabled=True,
                                      dtype=torch.get_autocast_gpu_dtype(),
                                      cache_enabled=torch.is_autocast_cache_enabled()):
             return f(*args, **kwargs)
