diff --git a/main.py b/main.py
index 5e03c1c..e11b7ea 100644
--- a/main.py
+++ b/main.py
@@ -914,7 +914,7 @@ if __name__ == "__main__":
 
             device = os.environ.get("CUDA_VISIBLE_DEVICES", "?")
             hostname = socket.gethostname()
-            ts = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+            ts = datetime.datetime.now(datetime.UTC).strftime("%Y-%m-%d %H:%M:%S")
             resp = requests.get("http://169.254.169.254/latest/meta-data/instance-id")
             print(
                 f"ERROR at {ts} on {hostname}/{resp.text} (CUDA_VISIBLE_DEVICES={device}): {type(err).__name__}: {err}",
